doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,724 @@
+
+When you create an application on Windows, you must link it as either a Windows
+or a Console application. When the application is launched from a commandline
+shell as a Windows application, the shell will immediately return to the
+foreground of the console, which means that any console output emitted by the
+process will be intermixed with the shell. However, if an application is linked
+as a Console application, and it's launched from the Start Menu, Run dialog, or
+any other context that's _not_ a console, then the OS will _automatically_
+create a console to host the commandline application. That means that briefly, a
+console window will appear on the screen, even if we decide that we just want to
+launch our application's window.
+
+This basically leaves us with two bad scenarios. Either we're a Console
+application, and a console window always flashes on screen for every
+non-commandline invocation of the Terminal, or we're a Windows application, and
+console output we log (including help messages) can get mixed with shell output.
+Neither of these are particularly good.
+
+`python` et. al. often ship with _two_ executables, a `python.exe` which is a
+Console application, and a `pythonw.exe`, which is a Windows application. This
+however has led to [loads of confusion](https://stackoverflow.com/a/30313091),
+and even with plentiful documentation, would likely result in users being
+confused about what does what. For situations like launching the Terminal in the
+CWD of `explorer.exe`, users would need to use `wtw.exe -d .` to prevent the
+console window from appearing. However, when calling Windows Terminal from a
+commandline environment, users who call `wtw.exe /?` would likely get unexpected
+behavior, because they should have instead called `wt.exe /?`.
+
+To avoid this confusion, I propose we follow the example of `msiexec /?`. This
+is a Windows application that uses a `MessageBox` to display its help text.
+While this is less convenient for users coming exclusively from a commandline
+environment, it's also the least bad option available to us.
+* It's less confusing than having control returned to the shell
+* It's not as bad as forcing the creation of a console window for
+  non-commandline launches.
+* There's precedent for this kind of dialog (we're not inventing a new pattern
+  here.)
+
+### What happens if `new-tab` isn't the first command?
+
+**TODO**: What should we do if we encounter the following?
+
+```sh
+wt.exe split-pane -v ; new-tab
+```
+
+In the future, maybe we could presume in this case that the commands are
+intended for the current Windows Terminal window, though that's not
+functionality that will arrive in 1.0. Even when sessions are supported like
+that, I'm not sure that when we're parsing a commandline, we'll be possible to
+know what session we're currently running in.That might make it challenging to
+dispatch this kind of command to "the current WT window".

```

Comment:
Should we need a `-r` option ("remote") to say "control a windows terminal instance?" with autodetection for the one it's running in? hmm